shell/platform/windows/platform_handler_unittests.cc

```diff
@@ -63,461 +69,334 @@ class TestPlatformHandler : public PlatformHandler {
                     std::unique_ptr<MethodResult<rapidjson::Document>>));
 };
 
-// Mock result to inspect results of PlatformHandler calls.
-class MockMethodResult : public MethodResult<rapidjson::Document> {
+// A test version of the private ScopedClipboard.
+class MockScopedClipboard : public ScopedClipboardInterface {
  public:
-  MOCK_METHOD1(SuccessInternal, void(const rapidjson::Document*));
-  MOCK_METHOD3(ErrorInternal,
-               void(const std::string&,
-                    const std::string&,
-                    const rapidjson::Document*));
-  MOCK_METHOD0(NotImplementedInternal, void());
+  MOCK_METHOD(int, Open, (HWND window), (override));
+  MOCK_METHOD(bool, HasString, (), (override));
+  MOCK_METHOD((std::variant<std::wstring, int>), GetString, (), (override));
+  MOCK_METHOD(int, SetString, (const std::wstring string), (override));
 };
 
-// A test version of system clipboard.
-class MockSystemClipboard {
- public:
-  void OpenClipboard() { opened = true; }
-  void CloseClipboard() { opened = false; }
-  bool opened = false;
-};
+std::string SimulatePlatformMessage(TestBinaryMessenger* messenger,
+                                    const uint8_t* message,
+                                    size_t message_size) {
```

Comment:
It feels like we can just use `strlen` for `message_size`, since that overhead shouldn't matter for unittests. Or even simpler, `std::string` for these strings.

Reply:
Sounds good, updated